### DISCUSSION TITLE
Graphics protocol not working with local clients?

### DISCUSSION THREAD

**Discussion by glyh** [kovidgoyal/kitty#7315] [Q&A] [answered]
I tried to invoke graphics protocol:
```
$ printf '\e_Gi=123,f=100,t=f;L2hvbWUvbHloL1BpY3R1cmVzL3NjcmVlbnNob3QucG5nCg==\e\\'
=123;EBADF:Failed to open file for graphics transmission with error: [2] No such file or directory^C
$ printf '\e_Gi=123,f=100,t=f;/home/lyh/Pictures/screenshot.png\e\\'
=123;EBADF:Failed to open file for graphics transmission with error: [2] No such file or directory^C
```
Both failed. The file do exist on my system and everything is running locally. What am I missing here? Thanks!

**Comment by kovidgoyal** [comment]
I suggest you use the bash script at https://sw.kovidgoyal.net/kitty/graphics-protocol/#a-minimal-example  and study its output to learn what you are doing wrong.

**Reply by kovidgoyal** [reply, marked as answer]
And just for posterity:
```
printf "\e_Gi=123,f=100,t=f,a=T;%s\e\\\\" "$(printf %s /path/to/image.png | base64 -w 0)"
```

**Reply by glyh** [reply]
I've already figured out how to use the protocol with buffer, here's my working example here: https://github.com/glyh/termcodes/blob/65db843f57b4e721a6ab7585494aae349162a51b/src/kitty.ysh#L15. 

 But it's not working with [local clients](https://sw.kovidgoyal.net/kitty/graphics-protocol/#local-client), and I'm asking specfically for that one.

**Reply by kovidgoyal** [reply]
My example is a local client and it works fine.

**Reply by glyh** [reply]
I tested again with 

```
printf '\e_Gi=123,f=100,t=f;2hvbWUvbHloL0RvY3VtZW50cy9DUy9vaWxzL3Rlcm1jb2Rlcy9yZXMvc2NyZWVuc2hvdC5wbmcK\e\\\\'
```

I can confirm the file coresponding to the base64 exists at 
`/home/lyh/Documents/CS/oils/termcodes/res/screenshot.png`
but kitty says 
` =123;EBADF:Failed to open file for graphics transmission with error: [2] No such file or directory^C`.

 Also the terminal response is thrown write at my shell, is there a way I can redirect it to a file descriptor? Thanks.

**Comment by kovidgoyal** [comment]
Use the actual command I posted replacing the path to the file in it.

**Reply by glyh** [reply]
Did you test it? On my end bash looks for more input: 

![image](https://github.com/kovidgoyal/kitty/assets/12870742/598efe1f-1976-42d9-a266-15098ada58b2)

**Reply by kovidgoyal** [reply]
You are missing the trailing double quote.

**Reply by glyh** [reply]
This seem to be working, and the issue I ran into is the path to format has a '\n' at the end for anyone interested. Thanks!
